commit 0cdc5e930a95ca2ec3f2fd9327da93a2fbaf8e97
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Sun Apr 10 23:58:43 2016 +0200

    Fix crash on getauxval call without HAVE_AUX_VECTOR
    
    	* sysdeps/generic/ldsodefs.h (struct rtld_global_ro)
    	[!HAVE_AUX_VECTOR]: Do not define _dl_auxv field.
    	* misc/getauxval.c (__getauxval) [!HAVE_AUX_VECTOR]: Do not go through
    	GLRO(dl_auxv) list.

diff --git a/misc/getauxval.c b/misc/getauxval.c
index e48f40f..6111376 100644
--- a/misc/getauxval.c
+++ b/misc/getauxval.c
@@ -30,9 +30,11 @@ __getauxval (unsigned long int type)
   else if (type == AT_HWCAP2)
     return GLRO(dl_hwcap2);
 
+#ifdef HAVE_AUX_VECTOR
   for (p = GLRO(dl_auxv); p->a_type != AT_NULL; p++)
     if (p->a_type == type)
       return p->a_un.a_val;
+#endif
 
   __set_errno (ENOENT);
   return 0;
diff --git a/sysdeps/generic/ldsodefs.h b/sysdeps/generic/ldsodefs.h
index 2733ac8..ddec0be 100644
--- a/sysdeps/generic/ldsodefs.h
+++ b/sysdeps/generic/ldsodefs.h
@@ -509,8 +509,10 @@ struct rtld_global_ro
   /* Mask for important hardware capabilities we honour. */
   EXTERN uint64_t _dl_hwcap_mask;
 
+#ifdef HAVE_AUX_VECTOR
   /* Pointer to the auxv list supplied to the program at startup.  */
   EXTERN ElfW(auxv_t) *_dl_auxv;
+#endif
 
   /* Get architecture specific definitions.  */
 #define PROCINFO_DECL

commit e1ef505659e9720ebdb116a21e91e0398cd411e5
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Mon Apr 11 10:27:02 2016 +0200

    Fix build with HAVE_AUX_VECTOR
    
    	* sysdeps/unix/sysv/linux/ldsodefs.h (HAVE_AUX_VECTOR): Define before
    	including <ldsodefs.h>.
    	* sysdeps/nacl/ldsodefs.h (HAVE_AUX_VECTOR): Likewise.

diff --git a/sysdeps/nacl/ldsodefs.h b/sysdeps/nacl/ldsodefs.h
index bc7723a..c6307a6 100644
--- a/sysdeps/nacl/ldsodefs.h
+++ b/sysdeps/nacl/ldsodefs.h
@@ -18,14 +18,14 @@
 
 #ifndef	_LDSODEFS_H
 
+/* We have the auxiliary vector.  */
+#define HAVE_AUX_VECTOR	1
+
 /* Get the real definitions.  */
 #include_next <ldsodefs.h>
 
 /* Now define our stuff.  */
 
-/* We have the auxiliary vector.  */
-#define HAVE_AUX_VECTOR	1
-
 /* Used by static binaries to check the auxiliary vector.  */
 extern void _dl_aux_init (ElfW(auxv_t) *av) internal_function;
 
diff --git a/sysdeps/unix/sysv/linux/ldsodefs.h b/sysdeps/unix/sysv/linux/ldsodefs.h
index 1f4d9e3..4c3676e 100644
--- a/sysdeps/unix/sysv/linux/ldsodefs.h
+++ b/sysdeps/unix/sysv/linux/ldsodefs.h
@@ -18,12 +18,12 @@
 
 #ifndef	_LDSODEFS_H
 
-/* Get the real definitions.  */
-#include_next <ldsodefs.h>
-
 /* We have the auxiliary vector.  */
 #define HAVE_AUX_VECTOR
 
+/* Get the real definitions.  */
+#include_next <ldsodefs.h>
+
 /* We can assume that the kernel always provides the AT_UID, AT_EUID,
    AT_GID, and AT_EGID values in the auxiliary vector from 2.4.0 or so on.  */
 #define HAVE_AUX_XID
